src/util.rs

```diff
@@ -152,6 +154,46 @@ impl From<anyhow::Error> for RpcError {
 /// A serializable result type for use in RPC responses.
 pub type RpcResult<T> = result::Result<T, RpcError>;
 
+#[derive(Debug, Display, Error)]
+pub(crate) enum BaoValidationError {
+    /// Generic io error. We were unable to read the data.
+    IoError(io::Error),
+    // /// The hash of the data does not match the hash of the outboard.
+    // HashMismatch,
+    // /// The size of the data does not match the size of the outboard.
+    // SizeMismatch,
+}
+
+impl From<io::Error> for BaoValidationError {
+    fn from(e: io::Error) -> Self {
+        BaoValidationError::IoError(e)
+    }
+}
+
+/// Validate that the data matches the outboard.
+pub(crate) fn validate_bao(
+    hash: Hash,
+    data_reader: impl Read + Seek,
+    outboard: Bytes,
+) -> result::Result<(), BaoValidationError> {
+    let hash = blake3::Hash::from(hash);
+    let outboard_reader = io::Cursor::new(outboard);
+    let mut decoder = abao::decode::Decoder::new_outboard(data_reader, outboard_reader, &hash);
+    // todo: expose chunk group size in abao, so people can allocate good sized buffers
+    let mut buffer = [0u8; 1024 * 16 + 4096];
```

Comment:
this is quite large for a stack array, probably makes more sense to use a Vec or BytesMut